packages/components/src/test/properties/inputs/UniquePropertyValuesSelector.test.tsx

```diff
@@ -1395,4 +1283,84 @@ describe("UniquePropertyValuesSelector", () => {
       });
     });
   });
+  describe("Items loader", () => {
+    it("does not load items when filter is empty", () => {
+      const spy = sinon.spy();
+      const itemsLoader = new ItemsLoader<string>(
+        () => {},
+        () => {},
+        () => spy(),
+        (option) => option,
+        [],
+      );
+      expect(itemsLoader.needsLoadingItems("")).to.be.false;
+    });
+
+    it("does not load items when loaded options matches the filter", async () => {
+      const getItemsStub = sinon.stub();
+      const itemsLoader = new ItemsLoader<string>(
+        () => {},
+        () => {},
+        () => getItemsStub(),
+        (option) => option,
+        [],
+      );
+
+      getItemsStub.callsFake(() => {
+        return {
+          options: Array.from({ length: UNIQUE_PROPERTY_VALUES_BATCH_SIZE }, () => "filterText"),
+          length: UNIQUE_PROPERTY_VALUES_BATCH_SIZE,
+          hasMore: true,
+        };
+      });
+      await itemsLoader.loadInitialItems();
+      expect(itemsLoader.needsLoadingItems("filterText")).to.be.false;
+    });
+
+    it("does not load items when hasMore is set to false", async () => {
+      const getItemsStub = sinon.stub();
+      const spy = sinon.spy();
+      const itemsLoader = new ItemsLoader<string>(
+        () => {},
+        () => spy,
+        () => getItemsStub(),
+        (option) => option,
+        [],
+      );
+
+      getItemsStub.callsFake(() => {
+        return {
+          options: Array.from({ length: UNIQUE_PROPERTY_VALUES_BATCH_SIZE }, () => "filterText"),
+          length: UNIQUE_PROPERTY_VALUES_BATCH_SIZE,
+          hasMore: false,
+        };
+      });
+      await itemsLoader.loadInitialItems();
+      await itemsLoader.loadItems("filterText");
+      expect(spy.calledOnce);
+    });
+
+    it("does not load items when items loader is disposed", async () => {
+      const getItemsStub = sinon.stub();
```

Comment:
```suggestion
        spy
```

Reply:
not sure what you mean by this change :D?

Reply:
Somehow this comment ended up on the wrong line. 😕  It was meant for `ItemsLoader` constructo call were you are passing `() => spy` or `() => getItemsStub()` you can pass `spy` or `getItemsStub` directly. Same for other tests in this file and `NavigationPropertyTargetsLoader` tests:

```ts
const itemsLoader = new ItemsLoader<string>(
        () => {},
        spy,
        getItemsStub,
        (option) => option,
        [],
      );
```

Reply:
changed